v-next/hardhat/test/internal/builtin-plugins/network-manager/network-manager.ts

```diff
@@ -359,6 +361,165 @@ describe("NetworkManagerImplementation", () => {
     });
   });
 
+  describe("connect when config has been extended by plugins", () => {
+    const networkConfigAddingPlugin: HardhatPlugin = {
+      id: "network-config-adding-plugin",
```

Comment:
This plugin simulates the equivalent of `hardhat-ledger` extending the Network config with additional properties.